test/unit-test/mqtt_agent_utest.c

```diff
@@ -81,14 +139,276 @@ int suiteTearDown( int numFailures )
+}
+
+void test_MQTTAgent_ResumeSession_publish_resend_success( void )
+{
+    bool sessionPresent = true;
+    MQTTStatus_t mqttStatus;
+    MQTTAgentContext_t mqttAgentContext;
+    AgentMessageInterface_t msgCtx = { 0 };
+    MQTTFixedBuffer_t networkBuffer = { 0 };
+    TransportInterface_t transportInterface = { 0 };
+    IncomingPublishCallback_t incomingCallback;
+    void * incomingPacketContext;
+    AgentMessageContext_t msg;
+    Command_t command;
+    MQTTPublishInfo_t args;
+    AckInfo_t ackInfo;
+
+    msgCtx.pMsgCtx = &msg;
+    msgCtx.send = mockSend;
+    msgCtx.recv = mockReceive;
+    msgCtx.releaseCommand = mockCommandRelease;
+    msgCtx.getCommand = mockGetCommand;
+    MQTT_Init_ExpectAnyArgsAndReturn( MQTTSuccess );
+
+    mqttStatus = MQTTAgent_Init( &mqttAgentContext, &msgCtx, &networkBuffer, &transportInterface, getTime, incomingCallback, incomingPacketContext );
+
+    command.pArgs = &args;
+    mqttAgentContext.mqttContext.nextPacketId = 1;
+
+    ackInfo.packetId = 1;
+    ackInfo.pOriginalCommand = &command;
+    mqttAgentContext.pPendingAcks[ 0 ] = ackInfo;
+
+    /* Check that publish ack is resent successfully when session resumes. */
+    MQTT_PublishToResend_IgnoreAndReturn( 1 );
+    MQTT_Publish_IgnoreAndReturn( MQTTSuccess );
+    MQTT_PublishToResend_IgnoreAndReturn( MQTT_PACKET_ID_INVALID );
+    mqttStatus = MQTTAgent_ResumeSession( &mqttAgentContext, sessionPresent );
+    TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
+}
+
+
+void test_MQTTAgent_ResumeSession_no_session_present( void )
 {
-    
+    bool sessionPresent = true;
+    MQTTStatus_t mqttStatus;
+    MQTTAgentContext_t mqttAgentContext;
+    AgentMessageInterface_t msgCtx = { 0 };
+    MQTTFixedBuffer_t networkBuffer = { 0 };
+    TransportInterface_t transportInterface = { 0 };
+    IncomingPublishCallback_t incomingCallback;
+    void * incomingPacketContext;
+    AgentMessageContext_t msg;
+    Command_t command;
+    AckInfo_t ackInfo, ackInfo_1;
+
+    msgCtx.pMsgCtx = &msg;
+    msgCtx.send = mockSend;
+    msgCtx.recv = mockReceive;
+    msgCtx.releaseCommand = mockCommandRelease;
+    msgCtx.getCommand = mockGetCommand;
+    MQTT_Init_ExpectAnyArgsAndReturn( MQTTSuccess );
+    mqttStatus = MQTTAgent_Init( &mqttAgentContext, &msgCtx, &networkBuffer, &transportInterface, getTime, incomingCallback, incomingPacketContext );
+
+    mqttAgentContext.mqttContext.nextPacketId = 1;
+    ackInfo.packetId = MQTT_PACKET_ID_INVALID;
+    mqttAgentContext.pPendingAcks[ 0 ] = ackInfo;
+    /* Check that acknowledgements with invalid packet Ids are not eligible for clearance. */
+    mqttStatus = MQTTAgent_ResumeSession( &mqttAgentContext, false );
+    TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
+    TEST_ASSERT_EQUAL_MEMORY( &ackInfo, &mqttAgentContext.pPendingAcks[ 0 ], sizeof( AckInfo_t ) );
+
+    ackInfo_1.packetId = 1;
+    ackInfo_1.pOriginalCommand = &command;
+    mqttAgentContext.pPendingAcks[ 1 ] = ackInfo_1;
```

Comment:
Do we even need `ackInfo_1`?
```suggestion
    mqttAgentContext.pPendingAcks[ 1 ].packetId = 1U;
    mqttAgentContext.pPendingAcks[ 1 ].pOriginalCommand = &command;
```